In [ ]:
%%sql
postgresql:///international_debt
select * from international_debt
limit 5;

5 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894453.700000003')),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440.100000001')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739336.899999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729.399999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620.100000001'))]

## 2. Finding the number of distinct countries
<p>From the first ten rows, we can see the amount of debt owed by Afghanistan</p>
<p>There are chances that the countries have debt in more than one debt indicator. So getting the distinct values in next cell. </p>

In [ ]:
%%sql
SELECT 
    COUNT(distinct country_name) AS total_distinct_countries
FROM international_debt;

 * postgresql:///international_debt
1 rows affected.


[(124,)]

## 3. Finding out the distinct debt indicators
<p>There are 124 countries present.</p> 
<p><code>indicator_name</code>means the reason of taking the debt.
<p><code>indicator_code</code> means different types of debt 
Knowing about these various debt indicators will make it easy to understand the areas in which a country can possibly be indebted to. </p>

In [ ]:
%%sql 
SELECT 
    COUNT(DISTINCT indicator_code) AS total_indicatos
FROM international_debt;

 * postgresql:///international_debt
1 rows affected.


[(25,)]

<p>There are total of 25 different indicators present in the table. In the next cell we get these different indicators</p>

In [ ]:
%%sql 
SELECT 
    DISTINCT indicator_code AS distinct_debt_indicators
FROM international_debt
ORDER BY distinct_debt_indicators;

 * postgresql:///international_debt
25 rows affected.


[('DT.AMT.BLAT.CD',),
 ('DT.AMT.DLXF.CD',),
 ('DT.AMT.DPNG.CD',),
 ('DT.AMT.MLAT.CD',),
 ('DT.AMT.OFFT.CD',),
 ('DT.AMT.PBND.CD',),
 ('DT.AMT.PCBK.CD',),
 ('DT.AMT.PROP.CD',),
 ('DT.AMT.PRVT.CD',),
 ('DT.DIS.BLAT.CD',),
 ('DT.DIS.DLXF.CD',),
 ('DT.DIS.MLAT.CD',),
 ('DT.DIS.OFFT.CD',),
 ('DT.DIS.PCBK.CD',),
 ('DT.DIS.PROP.CD',),
 ('DT.DIS.PRVT.CD',),
 ('DT.INT.BLAT.CD',),
 ('DT.INT.DLXF.CD',),
 ('DT.INT.DPNG.CD',),
 ('DT.INT.MLAT.CD',),
 ('DT.INT.OFFT.CD',),
 ('DT.INT.PBND.CD',),
 ('DT.INT.PCBK.CD',),
 ('DT.INT.PROP.CD',),
 ('DT.INT.PRVT.CD',)]

## 4. Totaling the amount of debt owed by the countries
<p>The financial debt of a particular country represents its economic state and it's development.</p>
<p>Total amount of debt (in USD) that is owed by the different countries is shown in the next cell. This will give us a sense of how the overall economy of the entire world is holding up.</p>

In [ ]:
%%sql
SELECT 
    ROUND(SUM(debt/1000000),3) AS total_debt
FROM international_debt; 

 * postgresql:///international_debt
1 rows affected.


[(Decimal('3079734.488'),)]

## 5. Country with the highest debt
<p>That is more than <em>3 million <strong>million</strong></em> USD, an amount which is really a big number. </p>
<p>Now that we have the exact total of the amounts of debt owed by several countries, let's now find out the country that owns the highest amount of debt along with the amount. 
<p>Debt is the sum of different debts owed by a country across several categories. This will help to understand more about the country in terms of its socio-economic scenarios.</p>

In [ ]:
%%sql
SELECT 
    country_name, ROUND(sum(debt),3) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt desc
limit 10;

 * postgresql:///international_debt
10 rows affected.


[('China', Decimal('285793494734.200')),
 ('Brazil', Decimal('280623966140.800')),
 ('South Asia', Decimal('247608723990.600')),
 ('Least developed countries: UN classification', Decimal('212880992791.900')),
 ('Russian Federation', Decimal('191289057259.200')),
 ('IDA only', Decimal('179048127207.300')),
 ('Turkey', Decimal('151125758035.300')),
 ('India', Decimal('133627060958.400')),
 ('Mexico', Decimal('124596786217.300')),
 ('Indonesia', Decimal('113435696693.500'))]

## 6. Average amount of debt across indicators
<p>So, it was <em>China</em>. More details of the China's debts can be found <a href="https://datatopics.worldbank.org/debt/ids/country/CHN">here</a>. </p>
<p>We now have a brief overview of the dataset and a few of its summary statistics. We already have an idea of the different debt indicators in which the countries owe their debts. 

<p>Now in next cell we have output of average debt in different debt indicators</p>

In [ ]:
%%sql
SELECT 
    indicator_code AS debt_indicator,
    indicator_name,
    AVG(debt) AS average_debt
FROM international_debt
GROUP BY debt_indicator,indicator_name
ORDER BY average_debt DESC
limit 10;

 * postgresql:///international_debt
10 rows affected.


[('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('5904868401.499193612')),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', Decimal('5161194333.812658349')),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', Decimal('2152041216.890243888')),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('1958983452.859836046')),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', Decimal('1803694101.963265321')),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', Decimal('1644024067.650806481')),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', Decimal('1223139290.398230108')),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', Decimal('1220410844.421518983')),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('1191187963.083064523')),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', Decimal('1082623947.653623188'))]

## 7. The highest amount of principal repayments
<p>We can see that the indicator <code>DT.AMT.DLXF.CD</code> tops the chart of average debt. This category means repayment of long term debts. More information about this category can be found <a href="https://datacatalog.worldbank.org/principal-repayments-external-debt-long-term-amt-current-us-0">here</a>. </p>

<p>We can clearly observe that there is a huge difference in the amounts of the indicators after the second one. Meaning that the first two indicators might be the most severe categories in which the countries owe their debts.</p>

<p>So in the next cell we investigate the long term debt indicator DT.AMT.DLXF.CD</p>

In [ ]:
%%sql
SELECT 
    country_name, 
    indicator_name
FROM international_debt
WHERE debt = (SELECT 
                 max(debt)
             FROM international_debt
             where indicator_code='DT.AMT.DLXF.CD');

 * postgresql:///international_debt
1 rows affected.


[('China', 'Principal repayments on external debt, long-term (AMT, current US$)')]

## 8. The most common debt indicator
<p>China has the highest amount of debt in the long-term debt (<code>DT.AMT.DLXF.CD</code>) category. This is verified by <a href="https://data.worldbank.org/indicator/DT.AMT.DLXF.CD?end=2018&most_recent_value_desc=true">The World Bank</a>. It is often a good idea to verify our analyses like this since it validates that our investigations are correct. </p>
<p>long-term debt is the topmost category when it comes to the average amount of debt. We will see in next cell if it is the most common indicator in which the countries owe their debt </p>

In [ ]:
%%sql

SELECT 
    indicator_code,
    COUNT(indicator_code) AS indicator_count
FROM international_debt
GROUP BY indicator_code
ORDER BY indicator_count DESC, indicator_code DESC
LIMIT 20;

 * postgresql:///international_debt
20 rows affected.


[('DT.INT.OFFT.CD', 124),
 ('DT.INT.MLAT.CD', 124),
 ('DT.INT.DLXF.CD', 124),
 ('DT.AMT.OFFT.CD', 124),
 ('DT.AMT.MLAT.CD', 124),
 ('DT.AMT.DLXF.CD', 124),
 ('DT.DIS.DLXF.CD', 123),
 ('DT.INT.BLAT.CD', 122),
 ('DT.DIS.OFFT.CD', 122),
 ('DT.AMT.BLAT.CD', 122),
 ('DT.DIS.MLAT.CD', 120),
 ('DT.DIS.BLAT.CD', 113),
 ('DT.INT.PRVT.CD', 98),
 ('DT.AMT.PRVT.CD', 98),
 ('DT.INT.PCBK.CD', 84),
 ('DT.AMT.PCBK.CD', 84),
 ('DT.INT.DPNG.CD', 79),
 ('DT.AMT.DPNG.CD', 79),
 ('DT.INT.PBND.CD', 69),
 ('DT.AMT.PBND.CD', 69)]

## 9. Other viable debt issues and conclusion
<p>There are a total of six debt indicators in which all the countries listed in our dataset have taken debt. The indicator <code>DT.AMT.DLXF.CD</code> is also there in the list. So, this gives us a clue that all these countries are suffering from a common economic issue</p>
<p>Now finding the debt indicators in which a country owes its highest debt. </p>
<p>So we took a look at debt owed by countries across the globe. We extracted a few summary statistics from the data and unraveled some interesting facts and figures.</p>

In [ ]:
%%sql
select country_name,
indicator_code,
max(debt) as maximum_debt
from international_debt
group by country_name,indicator_code
order by maximum_debt desc
limit 10;

 * postgresql:///international_debt
10 rows affected.


[('China', 'DT.AMT.DLXF.CD', Decimal('96218620835.699996948')),
 ('Brazil', 'DT.AMT.DLXF.CD', Decimal('90041840304.100006104')),
 ('China', 'DT.AMT.DPNG.CD', Decimal('72392986213.800003052')),
 ('Russian Federation', 'DT.AMT.DLXF.CD', Decimal('66589761833.5')),
 ('Turkey', 'DT.AMT.DLXF.CD', Decimal('51555031005.800003052')),
 ('South Asia', 'DT.AMT.DLXF.CD', Decimal('48756295898.199996948')),
 ('Brazil', 'DT.AMT.PRVT.CD', Decimal('43598697498.599998474')),
 ('Russian Federation', 'DT.AMT.DPNG.CD', Decimal('42800154974.900001526')),
 ('Brazil', 'DT.AMT.DPNG.CD', Decimal('41831444053.300003052')),
 ('Least developed countries: UN classification', 'DT.DIS.DLXF.CD', Decimal('40160766261.599998474'))]